src/libraries/Microsoft.Extensions.Logging/gen/LoggerMessageGenerator.Emitter.cs

```diff
@@ -0,0 +1,548 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+
+namespace Microsoft.Extensions.Logging.Generators
+{
+    public partial class LoggerMessageGenerator
+    {
+        internal class Emitter
+        {
+            // The maximum arity of LoggerMessage.Define.
+            private const int MaxLoggerMessageDefineArguments = 6;
+            private const int DefaultStringBuilderCapacity = 1024;
+
+            private readonly string _generatedCodeAttribute =
+                $"global::System.CodeDom.Compiler.GeneratedCodeAttribute(" +
+                $"\"{typeof(Emitter).Assembly.GetName().Name}\", " +
+                $"\"{typeof(Emitter).Assembly.GetName().Version}\")";
+            private readonly StringBuilder _builder = new StringBuilder(DefaultStringBuilderCapacity);
+
+            public string Emit(IReadOnlyList<LoggerClass> logClasses, CancellationToken cancellationToken)
+            {
+                _builder.AppendLine("// <auto-generated/>");
```

Comment:
Be sure to `.Clear()` the builder in between calls to `Emit`. Either at the beginning or at the end after calling `.ToString()`.

Reply:
Oh, I guess we don't re-use Emitter instances. No need to clear then.